lib/Data/Decimate.pm

```diff
@@ -0,0 +1,235 @@
+package Data::Decimate;
+
+use strict;
+use warnings;
+
+use 5.010;
+use Moose;
+
+use Date::Utility;
+
+use MooseX::Types::Moose qw(Int Num Str);
+use MooseX::Types -declare => [qw(
+        interval
+        )];
+
+use Moose::Util::TypeConstraints;
+use Time::Duration::Concise;
+
+subtype 'interval', as 'Time::Duration::Concise';
+coerce 'interval', from 'Str', via { Time::Duration::Concise->new(interval => $_) };
+
+=head1 NAME
+
+Data::Decimate - A module that allows to decimate a data feed. 
+
+=head1 VERSION
+
+Version 0.01
+
+=head1 SYNOPSIS
+
+  use Data::Decimate;
+
+  my $data_decimate = Data::Decimate->new;
+
+  my @data_feed = [
+        {epoch  => time,
+        ...},
+        {epoch  => time+1,
+        ...},
+        {epoch  => time+2,
+        ...},
+        ...
+  ];
+
+  my $output = $data_decimate->decimate(\@data_feed);
```

Comment:
I think you should give data-sample i.e. with real numbers. E.g. 15-data samples with epochs and spots, and the interval to 5 seconds, and 3 samples in the output.